Test/SqlDom/ParserErrorsTests.cs

```diff
@@ -4384,6 +4384,66 @@ public void CreateIndexStatementErrorTest()
                new ParserErrorInfo(47, "SQL46010", "col1"));
         }
 
+        /// <summary>
+        /// JSON Index error tests - ensure JSON Index syntax is rejected in older versions and malformed syntax produces appropriate errors
+        /// </summary>
+        [TestMethod]
+        [Priority(0)]
+        [SqlStudioTestCategory(Category.UnitTest)]
+        public void CreateJsonIndexStatementErrorTest()
+        {
+            // JSON Index syntax should not be supported in SQL Server versions prior to 2025 (TSql170)
+            // Test basic JSON Index syntax in older versions
+            ParserTestUtils.ErrorTest160("CREATE JSON INDEX idx1 ON table1 (jsonColumn)",
+                new ParserErrorInfo(7, "SQL46010", "JSON"));
+            ParserTestUtils.ErrorTest150("CREATE JSON INDEX idx1 ON table1 (jsonColumn)",
+                new ParserErrorInfo(7, "SQL46010", "JSON"));
+            ParserTestUtils.ErrorTest140("CREATE JSON INDEX idx1 ON table1 (jsonColumn)",
+                new ParserErrorInfo(7, "SQL46010", "JSON"));
+            ParserTestUtils.ErrorTest130("CREATE JSON INDEX idx1 ON table1 (jsonColumn)",
+                new ParserErrorInfo(7, "SQL46010", "JSON"));
+            ParserTestUtils.ErrorTest120("CREATE JSON INDEX idx1 ON table1 (jsonColumn)",
+                new ParserErrorInfo(7, "SQL46010", "JSON"));
+            ParserTestUtils.ErrorTest110("CREATE JSON INDEX idx1 ON table1 (jsonColumn)",
+                new ParserErrorInfo(7, "SQL46010", "JSON"));
+            ParserTestUtils.ErrorTest80("CREATE JSON INDEX idx1 ON table1 (jsonColumn)",
+                new ParserErrorInfo(7, "SQL46010", "JSON"));
+
+            // JSON Index with FOR clause in older versions
+            ParserTestUtils.ErrorTest160("CREATE JSON INDEX idx1 ON table1 (jsonColumn) FOR ('$.name')",
+                new ParserErrorInfo(7, "SQL46010", "JSON"));
+            
+            // JSON Index with WITH clause in older versions
+            ParserTestUtils.ErrorTest160("CREATE JSON INDEX idx1 ON table1 (jsonColumn) WITH (FILLFACTOR = 90)",
+                new ParserErrorInfo(7, "SQL46010", "JSON"));
+
+            // Test that UNIQUE and CLUSTERED/NONCLUSTERED are not allowed with JSON indexes in TSql170
+            TSql170Parser parser170 = new TSql170Parser(true);
+            ParserTestUtils.ErrorTest(parser170, "CREATE UNIQUE JSON INDEX idx1 ON table1 (jsonColumn)",
+                new ParserErrorInfo(19, "SQL46010", "INDEX"));
```

Comment:
why expected 19 here? running the test, the actual offset is 14 and the error is "Incorrect syntax near 'JSON'. at offset 14, line 1, column 15"